File: Reference Bots/Java/ReferenceBot/src/main/java/za/co/entelect/challenge/domain/state/OpponentCell.java
package za.co.entelect.challenge.domain.state;


import za.co.entelect.challenge.domain.command.Point;

public class OpponentCell {

    public boolean Damaged;

    public boolean Missed;

    public int X;

    public int Y;

    public Point getPoint() {

        return new Point(X, Y);
    }
}

File: Reference Bots/C#/ReferenceBot/Domain/State/OpponentCell.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using ReferenceBot.Domain.Command;

namespace ReferenceBot.Domain.State
{
    public class OpponentCell
    {
        [JsonProperty]
        public bool Damaged { get; set; }
        
        [JsonProperty]
        public bool Missed { get; set; }

        [JsonIgnore]
        public Point Point => new Point(X, Y);

        [JsonProperty]
        public int X { get; set; }

        [JsonProperty]
        public int Y { get; set; }
    }
}
